cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/converter/GaugeConverter.java

```diff
@@ -0,0 +1,25 @@
+package com.sap.cloud.cf.monitoring.java.converter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map.Entry;
+
+import com.codahale.metrics.Gauge;
+import com.sap.cloud.cf.monitoring.client.model.Metric;
+
+public final class GaugeConverter extends MetricConverter<Gauge> {
+
+    @Override
+    List<Metric> convertMetricEntry(Entry<String, Gauge> metricEntry, long timestamp) {
+        ArrayList<Metric> result = new ArrayList<>();
+        Object gaugeValue = metricEntry.getValue().getValue();
+        if (gaugeValue instanceof Number) {
+            Number number = (Number) gaugeValue;
+            result.add(buildCustomMetric(metricEntry.getKey() + ".value", number.doubleValue(), MetricType.GAUGE,
+                                         timestamp));
+        } else {

```

Comment:
Could be of course. I do not like try-catch for the program flow and another reason is the throwing an exception creates a new object.

Reply:
This code creates a new IllegalArgumentException object just in the line below. That is exactly what happens, if the cast would raise an ClassCastException. But in difference the ClassCastException would contain information, why it was raised. The current implementation just raises a rather unspecific runtime exception, that makes debugging harder.

Reply:
I really do not want to rely on exceptions for program flow and catching ClassCastException

Reply:
Currently you rely on throwing and catching an IllegalArgumentException. I do not see the benefits compared to the ClassCastException only the already mentioned drawbacks.

Can you at least create a message in the exception, that describes the cause?

Reply:
Reading the conversation here, I get the impression that you are talking about different things. What my understanding is:
* Karsten suggest to avoid the type check and the if-else-construct in favor of just attempting the cast to `Number`.
* If that cast would fail (because `gaugeValue` is not a number), a `ClassCastException` would be thrown including information on why that cast failed.
* In the current coding, that exception is replaced by an `IllegalArgumentException` that does not provide any additional information.

Given that, I don't see how this proposal is adding any additional complexity as far as program flow and exception handling is concerned. Abstract base class `MetricConverter` does not check for exceptions in `convert` which seems to indicate that conversion should always succeed. So, having a more descriptive exception would be benefical for the user. So, please follow the proposal or add a message in the `IllegalArgumentException` describing the reason for failure.